File: array/136-single-number/single-number.java
import java.util.HashMap;

class Solution {
    public int singleNumber(int[] nums) {
        int n = nums.length;

        // Declare the HasMap
        // And hash the given array:
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0; i<n; i++){
            int value = map.getOrDefault(nums[i], 0);
            map.put(nums[i], value+1);
        }

        
        
        //Find the single element and return the answer:
        for (int it : map.keySet()) {
            if (map.get(it) == 1)
                return it;
        }

        //This line will never execute
        //if the array contains a single element.
        return 0;

    }
}

//This will not work if contains negative numbers or a very large number.
//Use map data structure to hash the array elements.

File: array/136-single-number/single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        map<int, int> mpp;
        int n = nums.size();
        for(int i=0; i<n; i++){
            mpp[nums[i]]++;
        }

        for(auto it: mpp){
            if(it.second==1)
                return it.first;
        }
        return 0;
    }
};